### DISCUSSION TITLE
Module Development & Applying Theme

### DISCUSSION THREAD

**Discussion by Sharafudeen** [oqtane/oqtane.framework#1995] [General] [answered]
Hi,

I've two separate solutions for themes development. One is Bootswatch theme (downloaded from Oqtane GitHub sample) and second one is Patrix theme / custom theme (developed by me & Oqtane theme structures followed).

Now, I've a Oqtane Module called Country. I wanted to implement this module which can fit for both theme and it is totally different look & feel and some use cases even functionality may also differ.

For Bootswatch theme, I wanted to show Save & Cancel buttons are at bottom of the Modal popup. I want to show Grid filter as a panel or container (which is separated and top of the Grid).

For Patrix theme, I wanted to show Save & Cancel buttons are at top & bottom of the Modal popup. I want to show Grid filter in the grid column header.

What is the best way to do this?

Additionally.. How do I design login page in the theme module and how can I route or link from login button at the top of the navbar? I've tried but it is not working!

**Note:** Country module has Grid, Modal popup for CRUD operation, etc,.

**Comment by sbwalker** [comment, marked as answer]
Typically Modules and Themes are designed to be independent from one another... but If you want to have a single module that behaves differently based on specific themes then you would have to include some conditional logic in your module:

if (PageState.Page.ThemeType == "Patrix")
{
  ...
}

In regards to login, the Login component simply redirects a user to a "/login" route - which is a default page in all Oqtane installations. You can customize the Login page however you want (ie. assign a theme, add modules, etc...)
